prime-router/src/main/kotlin/cli/tests/Simulator.kt

```diff
@@ -369,6 +406,69 @@ class Simulator : CoolTest() {
 //            results += productionSimulation(environment, options)
 //            results += productionSimulation(environment, options)
 //        }
-        return teardown(results, elapsedTime)
+        return teardown(results, elapsedTime, startTime, options.asyncProcessMode)
+    }
+
+    /**
+     * Checks that at least the correct number of process records are in the Action table after running simulator.
+     * There may be more than expected if other tests occur at the same time, but there should not be fewer.
+     */
+    private suspend fun checkTimedResults(
+        expectedResults: Int,
+        afterDateTime: OffsetDateTime,
+        taskToCheck: TaskAction,
+        maxPollSecs: Int = 180,
+        pollSleepSecs: Int = 5
+    ): Boolean {
+        var resultsFound = 0
+
+        var timeElapsedSecs = 0
+        echo("Polling $taskToCheck records.  (Max poll time $maxPollSecs seconds)")
+        val actualTimeElapsedMillis = measureTimeMillis {
+            while (timeElapsedSecs <= maxPollSecs) {
+                if (outputToConsole) {
+                    for (i in 1..pollSleepSecs) {
+                        delay(1000)
+                        // Print out some contemplative dots to show we are waiting.
+                        print(".")
+                    }
+                    echo()
+                } else {
+                    delay(pollSleepSecs.toLong() * 1000)
+                }
+                timeElapsedSecs += pollSleepSecs
+
+                resultsFound = checkResultsQuery(afterDateTime, taskToCheck)
+
+                if (resultsFound >= expectedResults)
+                    break
+            }
+        }
+        echo("Polling for $taskToCheck records finished in ${actualTimeElapsedMillis / 1000 } seconds")
+
+        // verify results found count is greater than or equal to expected number
+        val passed = resultsFound >= expectedResults
+
+        if (passed) {
+            good("Found at least $expectedResults $taskToCheck records.")
+        } else {
+            bad("Did not find at least $expectedResults $taskToCheck records.")
+        }
+
+        return passed
+    }
+
+    private fun checkResultsQuery(afterDateTime: OffsetDateTime, actionType: TaskAction): Int {
+        var processActionsFound = 0
+        db = WorkflowEngine().db
+        db.transact { txn ->
+            val ctx = DSL.using(txn)
+            // find out how many process 'action' records were created after the start date
+            processActionsFound = ctx.selectFrom(Tables.ACTION)
+                .where(Tables.ACTION.ACTION_NAME.eq(actionType))
+                .and(Tables.ACTION.CREATED_AT.greaterOrEqual(afterDateTime))
+                .count()
+        }
```

Comment:
You know which receiver the data is destined for, so you can narrow your query to just that receiver.

PsuedoSQL might look like this (where $step = 'process', 'batch' or 'send'

```
select sum(item_count)
from report_file RF, action A
where 
   RF.action_id = A.action_id
   and RF.receiving_org = 'ignore'
   and RF.receiving_org_svc = 'EVERY_5_MINS'
   and RF.created_at > $afterDateTime
   and A.action_name = $step

```
Since that receiver is only used by Simulator, this should be accurate as long as no other simulator is running during that time.